packages/@lwc/engine/dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -38,3 +49,56 @@ export function deprecatedBuildCustomElementConstructor(
 
     return Ctor.CustomElementConstructor;
 }
+
+export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLElementConstructor {
+    const def = getComponentInternalDef(Ctor);
+
+    // generating the hash table for attributes to avoid duplicate fields and facilitate validation
+    // and false positives in case of inheritance.
+    const attributeToPropMap: Record<string, string> = create(null);
+    for (const propName in def.props) {
+        attributeToPropMap[getAttrNameFromPropName(propName)] = propName;
```

Comment:
can there be a case when getAttrNameFromPropName will return undefined? Wonder if we should do existence check prior writing into the map

Reply:
No, it's just a simple string transformation.